FirebaseSessions/Sources/SessionStartEvent.swift

```diff
@@ -74,8 +74,25 @@ class SessionStartEvent: NSObject, GDTCOREventDataObject {
       .performance = firebase_appquality_sessions_DataCollectionState_COLLECTION_SDK_NOT_INSTALLED
   }
 
+  deinit {
+    let garbage: [UnsafeMutablePointer<pb_bytes_array_t>?] = [
+      proto.session_data.session_id,
+      proto.session_data.first_session_id,
+      proto.application_info.app_id,
+      proto.application_info.session_sdk_version,
+      proto.application_info.device_model,
+      proto.application_info.apple_app_info.bundle_short_version,
+      proto.application_info.apple_app_info.mcc_mnc,
+    ]
+    for pointer in garbage {
+      nanopb_free(pointer)
+    }
+  }
+
   func setInstallationID(installationId: String) {
+    let oldID = proto.session_data.firebase_installation_id
     proto.session_data.firebase_installation_id = makeProtoString(installationId)
+    nanopb_free(oldID)
```

Comment:
what does these added lines do?

Reply:
Some of the fields in the nanopb proto are `malloc`ed byte buffers instead of integer, struct, or enum types, so they need to be manually freed.

Reply:
I'm keeping a reference to it around so I can pass it to the `nanopb_free` function. These are all `UnsafeMutablePointer`s, so they're references and the string buffers they point to aren't copied when assigning to a new variable.

I was worried something like
```swift
nanopb_free(proto.session_data.firebase_installation_id)
```
could cause issues with threaded access since the value held by the variable would be briefly garbage, though this implementation isn't thread-safe either.